### ISSUE TITLE
Requiring Allow in 405 responses (Trac #24)

### ISSUE THREAD

**Issue by mnot** [httpwg/httpbis-issues#24]
```text
#!html
            <p>In <a href="http://www.apps.ietf.org/rfc/rfc2616.html#sec-10.4.6">RFC 2616, section 10.4.6</a> 405 Method Not Allowed:</p>
            
            <blockquote><p>The method specified in the Request-Line is not allowed for the
            resource identified by the Request-URI. The response MUST include an
            Allow header containing a list of valid methods for the requested
            resource.</p></blockquote>
            
            <p>which has the effect of requiring that a server advertise all
            methods to a resource.  In some cases, method implementation is
            implemented across several (extensible) parts of a server and
            thus not known.  In other cases, it may not be prudent to tell
            an unauthenticated client all of the methods that might be
            available to other clients.</p>
```

Migrated from https://trac.ietf.org/ticket/24
```json
{
    "status": "closed",
    "changetime": "2008-04-10T06:16:48",
    "_ts": "1207808208000000",
    "description": "{{{\n#!html\n            <p>In <a href=\"http://www.apps.ietf.org/rfc/rfc2616.html#sec-10.4.6\">RFC 2616, section 10.4.6</a> 405 Method Not Allowed:</p>\n            \n            <blockquote><p>The method specified in the Request-Line is not allowed for the\n            resource identified by the Request-URI. The response MUST include an\n            Allow header containing a list of valid methods for the requested\n            resource.</p></blockquote>\n            \n            <p>which has the effect of requiring that a server advertise all\n            methods to a resource.  In some cases, method implementation is\n            implemented across several (extensible) parts of a server and\n            thus not known.  In other cases, it may not be prudent to tell\n            an unauthenticated client all of the methods that might be\n            available to other clients.</p>\n}}}",
    "reporter": "mnot@pobox.com",
    "cc": "",
    "resolution": "fixed",
    "time": "2007-12-20T03:18:55",
    "component": "p2-semantics",
    "summary": "Requiring Allow in 405 responses",
    "priority": "",
    "keywords": "",
    "milestone": "unassigned",
    "owner": "",
    "type": "design",
    "severity": ""
}
```

**Comment by mnot** [comment]
Trac update at 20071220T03:19:09: mnot@pobox.com commented:

> Proposal: Change the MUST to MAY in 10.4.6.

**Comment by mnot** [comment]
Trac update at 20080104T06:17:45: 
* mnot@pobox.com changed component from "" to "semantics"
* mnot@pobox.com changed milestone from "" to "unassigned"
* mnot@pobox.com changed owner from "" to ""
* mnot@pobox.com changed version from "" to "d00"

**Comment by mnot** [comment]
Trac update at 20080317T09:56:53: mnot@pobox.com commented:

> Proposal: 
> 
>  * In p2 10.1, change "The actual set of allowed methods is defined by the origin server at the time of each request."  to  "The actual set of allowed methods is defined by the origin server at the time of each request, and may not necessarily include all (or any) methods that the server would actually allow in a request if presented."  (with normal editorial discretion)
> 
>  * In p2 10.1, remove "However, the indications given by the Allow header field value SHOULD be followed."
>

**Comment by mnot** [comment]
Trac update at 20080317T12:17:50: 
* julian.reschke@greenbytes.de changed attachment from "" to "i24.diff"
* julian.reschke@greenbytes.de commented:

> Proposed change (see http://lists.w3.org/Archives/Public/ietf-http-wg/2008JanMar/0609.html)

**Comment by mnot** [comment]
Trac update at 20080404T00:19:19: mnot@pobox.com commented:

> Updated proposal:
>   
> * In the definition of Allow, change:
>  The Allow entity-header field lists the set of methods supported by the resource identified by the Request-URI.
> to
>  The Allow entity-header field advertises a set of methods as supported by the resource identified by the Request-URI.
> 
> * And, remove:
>  This field cannot prevent a client from trying other methods. However, the indications given by the Allow header field value SHOULD be followed.
> 
>

**Comment by mnot** [comment]
Trac update at 20080410T06:16:48: 
* julian.reschke@gmx.de commented:

> Fixed in [240]:
> 
> Resolve #24: relax requirements for contents of Allow header (closes #24).


* julian.reschke@gmx.de changed resolution from "" to "fixed"
* julian.reschke@gmx.de changed status from "new" to "closed"
